### ISSUE TITLE
flycheck does not handle cppcheck messages that do not reference a line

### ISSUE THREAD

**Issue by dholm** [flycheck/flycheck#187]
Hi,
While working on some C code I found an issue with cppcheck support in flycheck. cppcheck can generate messages that do not reference any lines, i.e. a general message referencing the entire file. In this case it was a file with two many alternate configurations wrapped in ifdefs.

I tried fixing this issue myself but I am still a Lisp noob so I wasn't able to find a good solution to the problem.

Here is a snippet that reproduces this issue, tested with cppcheck 1.60.1.

``` c
void f(void)
{
#if defined(ALTERNATIVE_1)
#elif defined(ALTERNATIVE_2)
#elif defined(ALTERNATIVE_3)
#elif defined(ALTERNATIVE_4)
#elif defined(ALTERNATIVE_5)
#elif defined(ALTERNATIVE_6)
#elif defined(ALTERNATIVE_7)
#elif defined(ALTERNATIVE_8)
#elif defined(ALTERNATIVE_9)
#elif defined(ALTERNATIVE_10)
#elif defined(ALTERNATIVE_11)
#elif defined(ALTERNATIVE_12)
#endif
}
```

Message from flycheck when opening this snippet in Emacs:

```
Failed to parse errors from checker c/c++-cppcheck in output: <?xml version="1.0" encoding="UTF-8"?>                                                                                                                                           
<results version="2">                                                                                                                                                                                                                          
  <cppcheck version="1.60.1"/>                                                                                                                                                                                                                 
  <errors>                                                                                                                                                                                                                                     
  <error id="toomanyconfigs" severity="information" msg="Too many #ifdef configurations - cppcheck only checks 12 configurations. Use --force to check all configurations. For more details, use --enable=information." verbose="The checking \
of the file will be interrupted because there are too many #ifdef configurations. Checking of all #ifdef configurations can be forced by --force command line option or from GUI preferences. However that may increase the checking time. For\
 more details, use --enable=information.">                                                                                                                                                                                                     
  </error>                                                                                                                                                                                                                                     
  </errors>                                                                                                                                                                                                                                    
</results>                                                                                                                                                                                                                                     

Error: Wrong type argument: listp, "                                                                                                                                                                                                           
  "
```

Since the message references the entire file (or is an internal error, whichever way you want to interpret it) it doesn't include the "location" XML tag so flycheck is unable to parse it.

The result of running cppcheck --quiet --xml-version=2 --inline-suppr test.c is:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<results version="2">
  <cppcheck version="1.60.1"/>
  <errors>
  <error id="toomanyconfigs" severity="information" msg="Too many #ifdef configurations - cppcheck only checks 12 configurations. Use --force to check all configurations. For more details, use --enable=information." verbose="The checking of the file will be interrupted because there are too many #ifdef configurations. Checking of all #ifdef configurations can be forced by --force command line option or from GUI preferences. However that may increase the checking time. For more details, use --enable=information.">
  </error>
  </errors>
</results>
```
